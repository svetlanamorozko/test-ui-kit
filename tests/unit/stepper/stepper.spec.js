import { shallow } from 'enzyme';
import React from 'react';
import Stepper from '../../../components/stepper/stepper';

const event = {
  preventDefault: jest.fn(),
};

describe('Stepper', () => {
  describe('#render()', () => {
    it('should render component', () => {
      const wrapper = shallow(
        <Stepper
          initValue={2}
          minValue={1}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render with disabled decrement button', () => {
      const wrapper = shallow(
        <Stepper
          initValue={1}
          minValue={1}
        />
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('.ui-stepper__button-decrease').hasClass('ui-stepper__button-disabled')).toBe(true);
    });

    it('should render with disabled increment button', () => {
      const wrapper = shallow(
        <Stepper
          initValue={3}
          maxValue={3}
        />
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('.ui-stepper__button-increase').hasClass('ui-stepper__button-disabled')).toBe(true);
    });

    it('should reset current value if initValue prop changes (2 -> 4)', () => {
      const wrapper = shallow(
        <Stepper
          initValue={2}
        />
      );

      expect(wrapper.find('.ui-stepper__value').text()).toEqual('2');
      wrapper.setProps({ initValue: 4 });
      expect(wrapper.find('.ui-stepper__value').text()).toEqual('4');
    });

    it('should not reset current value if props change but initValue stays the same (2 -> 2)', () => {
      const wrapper = shallow(
        <Stepper
          initValue={2}
        />
      );

      wrapper.setProps({ initValue: 2 });
      expect(wrapper.find('.ui-stepper__value').text()).toEqual('2');
    });

    it('should decrement value (2 -> 1)', () => {
      const wrapper = shallow(
        <Stepper
          initValue={2}
          maxValue={5}
          minValue={1}
        />
      );

      wrapper.find('.ui-stepper__button-decrease').simulate('click', event);
      expect(wrapper.find('.ui-stepper__value').text()).toEqual('1');
    });

    it('should not decrement value (1 -> 1)', () => {
      const wrapper = shallow(
        <Stepper
          initValue={1}
          maxValue={5}
          minValue={1}
        />
      );

      wrapper.find('.ui-stepper__button-decrease').simulate('click', event);
      expect(wrapper.find('.ui-stepper__value').text()).toEqual('1');
    });

    it('should increment value (2 -> 3)', () => {
      const wrapper = shallow(
        <Stepper
          initValue={2}
          maxValue={5}
          minValue={1}
        />
      );

      wrapper.find('.ui-stepper__button-increase').simulate('click', event);
      expect(wrapper.find('.ui-stepper__value').text()).toEqual('3');
    });

    it('should not increment value (5 -> 5)', () => {
      const wrapper = shallow(
        <Stepper
          initValue={5}
          maxValue={5}
          minValue={1}
        />
      );

      wrapper.find('.ui-stepper__button-increase').simulate('click', event);
      expect(wrapper.find('.ui-stepper__value').text()).toEqual('5');
    });

    it('should call callback after value changed', () => {
      const callback = jest.fn();
      const wrapper = shallow(
        <Stepper
          initValue={1}
          maxValue={5}
          minValue={1}
          onChange={callback}
        />
      );

      wrapper.find('.ui-stepper__button-increase').simulate('click', event);
      expect(callback).toHaveBeenCalledTimes(1);
      expect(callback).toHaveBeenCalledWith(2);
    });

    it('should not call callback if value is not changed', () => {
      const callback = jest.fn();
      const wrapper = shallow(
        <Stepper
          initValue={5}
          maxValue={5}
          minValue={1}
          onChange={callback}
        />
      );

      wrapper.find('.ui-stepper__button-increase').simulate('click', event);
      expect(callback).toHaveBeenCalledTimes(0);
    });
  });
});
