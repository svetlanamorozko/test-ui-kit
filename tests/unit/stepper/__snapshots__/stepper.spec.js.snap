// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper #render() should render component 1`] = `
<div
  className="ui-stepper"
>
  <input
    name="input"
    type="hidden"
    value={2}
  />
  <div
    className="ui-stepper__content"
  >
    <button
      className="ui-stepper__button ui-stepper__button-decrease"
      onClick={[Function]}
      type="button"
    />
    <div
      className="ui-stepper__value"
    >
      2
    </div>
    <div
      className="ui-stepper__shadow"
    />
    <button
      className="ui-stepper__button ui-stepper__button-increase"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Stepper #render() should render with disabled decrement button 1`] = `
<div
  className="ui-stepper"
>
  <input
    name="input"
    type="hidden"
    value={1}
  />
  <div
    className="ui-stepper__content"
  >
    <button
      className="ui-stepper__button ui-stepper__button-decrease ui-stepper__button-disabled"
      onClick={[Function]}
      type="button"
    />
    <div
      className="ui-stepper__value"
    >
      1
    </div>
    <div
      className="ui-stepper__shadow"
    />
    <button
      className="ui-stepper__button ui-stepper__button-increase"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Stepper #render() should render with disabled increment button 1`] = `
<div
  className="ui-stepper"
>
  <input
    name="input"
    type="hidden"
    value={3}
  />
  <div
    className="ui-stepper__content"
  >
    <button
      className="ui-stepper__button ui-stepper__button-decrease"
      onClick={[Function]}
      type="button"
    />
    <div
      className="ui-stepper__value"
    >
      3
    </div>
    <div
      className="ui-stepper__shadow"
    />
    <button
      className="ui-stepper__button ui-stepper__button-increase ui-stepper__button-disabled"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
